build-support/bin/check_banned_imports.sh

```diff
@@ -3,21 +3,21 @@
 PYTHON_FILES="$(find src tests pants-plugins examples contrib -name '*.py')"
 RUST_FILES="$(find src/rust/engine -name '*.rs')"
 
-bad_files="$(echo ${PYTHON_FILES} | xargs grep -l "^import subprocess$")"
+bad_files="$(echo "${PYTHON_FILES}" | xargs grep -l "^import subprocess$")"
```

Comment:
This actually breakingly changes the behaviour. Right now we rely on file names not having spaces in for this to work correctly, but with this change if more than one file is matched, the grep will not find anything.

If you want to do proper quoting here, you'd probably need to use some kind of array for $PYTHON_FILES. Or we could just convert this whole file to Python or something.

Same throughout this file.

Reply:
Good catch! I would love to rewrite to Python, but tried to not get distracted so simply fixed the issue here.

Reply:
Just kidding :/ this solution does not work like I thought. It now creates an array with one element: the string with the list of all the files separated by `\n`.

I'm going to revert all changes to this file and regroup in a followup PR. I need to make some changes anyways for Python 3 support.